### DISCUSSION TITLE
结构方程模型：罄竹难书 | /cn/2008/01/why-i-hate-structural-equation-models/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#370] [Comments]
https://yihui.org/cn/2008/01/why-i-hate-structural-equation-models/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
PLS比起协方差建模方法是要稍微好一点点，但PLS同样也仅仅只利用了协方差的信息，相比之下其优点在于计算的复杂性大大降低了，求特征向量/特征根的问题我们完全是有把握的，而SEM协方差建模的方法中，求那个目标函数的极小值根本就是非常扯的事情，难以想象。

> ——原帖发布于 2008-01-11 09:48:51

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
我想它popular的一点重要原因就是可以任人摆布，要想做出一个SEM不支持的理论，难

> ——原帖发布于 2008-01-11 09:43:22

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
No problem

> Originally posted on 2008-01-11 09:21:38

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
多谢李老师的教诲！

1、我一直都在想的问题是，若只用协方差的信息建模，那么意思就是说，只要两批样本的协方差矩阵相同，那么在结构方程模型的眼中，这两批样本就没有任何区别，我觉得这是不合理的。另外，我从您说的ADF模型中看出您似乎有点误解我的意思了，我不是说数据是否服从正态分布的问题（这也是问题之一），而是说对数据的信息利用充分程度。另外，我不知道协方差矩阵是不是充分统计量（初步感觉应该不是），还望老师指点。

2、这一点也要请教老师，应用者通常有没有注意两个问题：（1）对于如此高度复杂的优化问题，是否多次更改迭代初始值，让程序多次计算并检验计算结果的稳定性；（我所了解的一位老师就完全没有意识到这个问题的重要性，这让我非常吃惊）（2）有没有尝试类似leave-one-out之类的算法，随意删掉一条样本记录，再重新计算，看结果是否有大的波动。我认为在SEM的计算过程中，这两点是一定要做的，而我所读到的论文似乎都很少提及这些事情，因此我很怀疑那些看似漂亮的结果背后隐藏着多大的不稳定性。

3、我的意思是，SEM的思想源自一个假设检验，那么这就注定我们只有把握拒绝这个模型，而当我们作出“接受某个模型”的决定时，我们是很难意识到犯第II类错误的危险的。至于RMSEA之类的辅助指标，基本上都取决于怎样定义S和Sigma之间的差异，而这个差异要怎样小才算小，多大又算大？也是没有一定之规，RMSEA<0.05就是好模型吗？不一定。

至于您提到的因子命名的问题，就是更麻烦的问题了，几乎就没办法讲理了……我甚至想知道有没有SEM做不出来的理论（东家一套命名，西家一套命名，大家各自都能得到SEM的支持）。

以上四个问题中，有两个是Norman Cliff(1983)提到过的。我觉得对于SEM的研究者来说，这篇论文还是非常有价值的。

> ——原帖发布于 2008-01-15 20:06:17

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
从您的回复中我的确又学到了一些新的东西（准确地说是视角），所以“教诲”也不为过啦，呵呵

1、这么想倒也有道理，统计的祖宗之一Pearson做的工作确实是在计算前四阶矩。

2、我这么看重计算的问题，是因为我感觉国内的统计氛围中，对统计计算忽视得太严重了，尤其是过于依赖软件，我是非常支持开源软件的，因此对于无法看到源代码的软件始终有种不信任的感觉，把具体的计算过程封闭起来不让别人看到，这样谁都无法了解计算中是否存在bug。R里面有sem包，所以我可以从其源代码中看见John Fox用的是nlm()函数来完成最小化，而LISREL用的是什么方法来最小化目标函数的呢，官方网站上似乎没有提及这件事。

3、RMSEA不是SQRT[(Chi^2 - df / (N - 1))/df]么？这里面看不出它有上界吧？因为没有上界（参照物），所以就无法判断多小算作足够小。统计里面有些指标是有范围的，像您提到的R^2，如果不钻牛角尖，我们还是可以讲0.9算大0.1算小的，因为它的取值在[0，1]之间。而且，在ML估计的情况下，即使Chi^2=0、RMSEA=0，也反推不出S=Sigma（这二者之间没有if and only if的关系）。

4、我个人的意思是，模型的灵活性是好的，但SEM也未免太灵活了。

> ——原帖发布于 2008-01-16 10:58:22

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
好的，等有空我争取读一读您提到的文章，也期待你们的作品：）

非常感谢您的意见，难得我的Blog上有如此充分的交流 :grin:

> ——原帖发布于 2008-01-16 21:05:51

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
概率论或者数理统计的书。

我记得你的邮箱，你应该没认错人：）

> ——原帖发布于 2008-07-01 02:54:40

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
不要受我蛊惑，我发牢骚只是SEM这个东西几乎人人都说好，越说越神奇，这就有问题了。自己好好琢磨琢磨吧。

> ——原帖发布于 2009-12-08 06:20:03

[a comment by an automated account is not shown]

**Comment by hezhan9986** [comment]
我的天哪，我完全支持谢老师的观点！真的是完全说出了我的心声！最近在做研究时对SEM产生了深深的思考：我为什么要做它，它能解决什么实际问题？
感谢谢老师的博文，让我豁然开朗！谢谢！非常感谢！相见恨晚！

**Reply by yihui** [reply]
年少无知的我凶狠起来我自己都害怕，哈哈。后面我还发过好几篇牢骚：

- https://yihui.org/cn/2008/05/no-more-consultation-about-structural-equation-models/
- https://yihui.org/cn/2008/09/ridiculous-to-use-pls-to-solve-sem/
- https://yihui.org/cn/2009/03/blind-faith-in-structural-equation-models/
- https://yihui.org/cn/2017/06/sem-articles/
